internal/service/searchindex/resource_search_index.go

```diff
@@ -281,9 +281,8 @@ func resourceUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 				"cluster_name": clusterName,
 				"index_id":     indexID,
 			}))
-			resourceDelete(ctx, d, meta)
 			d.SetId("")
-			return diag.FromErr(fmt.Errorf("error creating index in cluster (%s): %s", clusterName, err))
```

Comment:
if we don't delete it, should we keep the id and don't set it to "" ?

Reply:
nice catch! fixed